### PR TITLE
Check ready state of document in public pages

### PR DESCRIPTION
**Pull request by ykzts**
`main()` will wait for polyfills loading. Therefore, there are times when it is executed after loading the HTML document.

Check at the value of [`document.readyState`](https://html.spec.whatwg.org/multipage/dom.html#current-document-readiness) to see if the loading is completed.

### CODE REVIEW

File: app/javascript/packs/public.js [hunk 1]
```diff
@@ -21,23 +21,29 @@ function main() {
...
-      content.textContent = relativeFormat.format(datetime);;
-    });
+  [].forEach.call(document.querySelectorAll('time.time-ago'), (content) => {
+    const datetime = new Date(content.getAttribute('datetime'));
+    content.textContent = relativeFormat.format(datetime);;
   });
+}
+
+function main() {
+  if (document.readyState === 'complete') {
```
Comment (yumetodo) [inline app/javascript/packs/public.js]: Which is better `document.readyState === 'complete'` or `document.readyState !== '"loading'`?

Comment (ykzts) [inline app/javascript/packs/public.js]: Oh, I'm sorry. I haven't seen this comment. I  just made similar changes. c538e82

File: app/javascript/packs/public.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ function loaded() {
 }
 
 function main() {
-  if (document.readyState === 'complete') {
+  if (['interactive', 'complete'].includes(document.readyState)) {
```
Comment (yumetodo) [inline app/javascript/packs/public.js]: Why don't you write `document.readyState !== 'loading'` instead of calling `Array.prototype.includes()`?

Comment (ykzts) [inline app/javascript/packs/public.js]: Because want to express state explicitly.

**Review by clworld — APPROVED** [review]
Looks good to me.

**Pull request merged**